mxcubeweb/routes/ra.py

```diff
@@ -186,14 +181,21 @@ def request_control_response():
         data = request.get_json()
         new_op = observer_requesting_control()
 
-        # Request was denied
-        if not data["giveControl"]:
-            remain_observer(new_op, data["message"])
-        else:
+        if data["giveControl"]:
+            # Request approved
             toggle_operator(new_op.username, data["message"])
-
-        new_op.requests_control = False
-        app.usermanager.update_user(new_op)
+        else:
+            # Request denied
+            new_op.requests_control = False
+            new_op.requests_control_msg = None
+            app.usermanager.update_user(new_op)
+            server.emit(
+                "userChanged",
+                data["message"],
+                room=new_op.socketio_session_id,
+                namespace="/hwr",
+            )
```

Comment:
The fix for the "asking for control" dialog not hiding after being denied, together with the new condition in the `userChanged` message handler in `serverIO`.